bowling-ball/Game.cs

```diff
@@ -2,16 +2,63 @@
 
 namespace BowlingBall
 {
+    
     public class Game
     {
-        public void Roll(int pins)
-        {
-            throw new NotImplementedException();
+        private int[] Rolls = new int[21];
+        private int currentRoll;
+        public void Roll(int pins)
+        {
+            Rolls[currentRoll++] = pins;
         }
 
-        public int GetScore()
+        public int GetScore()
+        {
+            int score = 0;
+            int roll = 0;
+            for (int frame = 1; frame <= 10; frame++)
+            {
+                if(IsStrike(roll))// strike bonus 2 next rolls so first check for strike.
+                {
+                    score += 10 + StrikeBonus(roll);
+                    roll++;
+                }
+                else if (IsSpare(roll))
+                {
+                    score += 10 + SpareBonus(roll);
+                    roll += 2;
+                }
+                    
+                else
+                {
+                    score += SumOfBallInFrame(roll);
+                    roll += 2;
+                }
+
+                
+            }
+            return score;
+        }
+
+        private bool IsSpare(int roll)
+        {
+            return Rolls[roll] + Rolls[roll + 1] == 10;

```

Comment:
It is recommended to check whether the collection has same or more number of elements, before accessing the specific element. This will help to avoid index out of range exception.